torchdata/nodes/samplers/multi_node_round_robin_sampler.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import copy
+from typing import Any, Dict, Mapping, Optional
+
+from torchdata.nodes.base_node import BaseNode, T
+from torchdata.nodes.samplers.stop_criteria import StopCriteria
+
+
+class MultiNodeRoundRobinSampler(BaseNode[T]):
```

Comment:
What is the expected behavior when doing many epochs over all the datasets? Would every epoch be the same?

Reply:
Yes, every epoch would be the same. Added a new test for that. https://github.com/pytorch/data/blob/5f6370502f38c7a4211af6e54123f060e0f7f151/test/nodes/test_multi_node_round_robin_sampler.py#L177